TebLocalPlannerROS:

    xy_goal_tolerance: .5

    odom_topic: odom
    map_frame: /map

    # Trajectory

    teb_autosize: True
    dt_ref: 0.3
    dist_step: 0.0
    dt_hysteresis: 0.1
    global_plan_overwrite_orientation: True
    max_global_plan_lookahead_dist: 3.0
    feasibility_check_no_poses: 5
    detour_threshold: -.1

    # Robot

    max_vel_x: 0.4 # Changed!
    max_vel_x_backwards: 0.2
    max_vel_theta: 5.0
    acc_lim_x: 1.0
    acc_lim_theta: 2.0
    min_turning_radius: 0.0
    footprint_model: # types: "point", "circular", "two_circles", "line", "polygon"
        type: "point"
        radius: 0.2 # for type "circular"
        line_start: [-0.3, 0.0] # for type "line"
        line_end: [0.3, 0.0] # for type "line"
        front_offset: 0.2 # for type "two_circles"
        front_radius: 0.2 # for type "two_circles"
        rear_offset: 0.2 # for type "two_circles"
        rear_radius: 0.2 # for type "two_circles"
        vertices: [ [0.25, -0.05], [0.18, -0.05], [0.18, -0.18], [-0.19, -0.18], [-0.25, 0], [-0.19, 0.18], [0.18, 0.18], [0.18, 0.05], [0.25, 0.05] ] # for type "polygon"
    max_vel_y: 0.4
    acc_lim_y: 1.0
    # GoalTolerance
    free_goal_vel: False

    allow_init_with_backwards_motion: false #doesn't seem to work


    # Obstacles

    min_obstacle_dist: 0.21
    include_costmap_obstacles: True
    costmap_obstacles_behind_robot_dist: 1.0
    obstacle_poses_affected: 30 #only has effect if using legacy obstacle mode
    #inflation_dist: .6
    costmap_converter_plugin: ""
    costmap_converter_spin_thread: True
    costmap_converter_rate: 5

    # Optimization

    no_inner_iterations: 5
    no_outer_iterations: 4
    optimization_activate: True
    optimization_verbose: False
    penalty_epsilon: 0.1
    weight_max_vel_x: 2
    weight_max_vel_y: 1
    weight_max_vel_theta: 1
    weight_acc_lim_x: 1
    weight_acc_lim_y: 1
    weight_acc_lim_theta: 1
    weight_kinematics_nh: 8 #1000
    weight_kinematics_forward_drive: 1000 # changed: was 1. set to ~1000 to prevent driving backwards
    weight_kinematics_turning_radius: 1
    weight_optimaltime: 1
    weight_obstacle: 500
    weight_dynamic_obstacle: 10 # not in use yet
    selection_alternative_time_cost: False # not in use yet

    # Homotopy Class Planner

    enable_homotopy_class_planning: True
    enable_multithreading: True #True
    simple_exploration: True #was False
    max_number_classes: 4
    roadmap_graph_no_samples: 15
    roadmap_graph_area_width: 6
    h_signature_prescaler: 1 #was 0.5
    h_signature_threshold: 0.1 #was .1
    obstacle_keypoint_offset: 0.1
    obstacle_heading_threshold: 0.45 #was 0.45
    viapoints_all_candidates: False
    visualize_hc_graph: True # was False
    
    selection_cost_hysteresis: .7  # new_cost must be < old_cost * this
